### PR TITLE
Update mortgage example using TaskGraph API.

### PR DESCRIPTION
**Pull request by avolkov1**
In the `mortgage_run_workflow_daskdistrib.py` and the notebook, because of deepcopy of conf when I was passing the dask client through the conf:
```
    mortgage_workflow_runner_task = {
        TaskSpecSchema.task_id:
            MortgageTaskNames.dask_mortgage_workflow_runner_task_name,
        TaskSpecSchema.node_type: 'DaskMortgageWorkflowRunner',
        TaskSpecSchema.conf: {
            'mortgage_run_params_dict_list': mortgage_run_params_dict_list,
            'client': client,  # DASK CLIENT CANNOT BE PICKLED I.E. DEEPCOPY BREAKS
            'use_rmm': use_rmm,
            'filter_dask_logger': filter_dask_logger,
        },
        TaskSpecSchema.inputs: [],
        TaskSpecSchema.filepath: mortgage_lib_module
    }
```

I was getting:
```
Traceback (most recent call last):
  File "/home/avolkov/projects/gitrepos/cuquant/gquant/dataframe_flow/task.py", line 28, in __init__
    copy.deepcopy(task_spec[TaskSpecSchema.conf])
  File "/home/avolkov/progs/python_installs/miniconda3/envs/py36-rapids/lib/python3.6/copy.py", line 150, in deepcopy
    y = copier(x, memo)
    . . .
TypeError: can't pickle _asyncio.Future objects
```

I changed the code by passing the dask scheduler address, just a string, and
creating a new client.

That caused another error:
```
  File "mortgage_gquant_plugins.py", line 1341, in process
    dmat_delayed_persist = dmat_delayed.persist()
ValueError: Inputs contain futures that were created by another client.
```

I could implement this by having the dask client be an output from the first
task in the task-graph, and passed as an input to the next task. The simplest thing
to do though is just to not deepcopy the conf fields.

### CODE REVIEW

File: notebook/mortgage_e2e_gquant/mortgage_gquant_plugins.py [hunk 1]
```diff
@@ -772,14 +772,15 @@ def mortgage_gquant_run(run_params_dict):
...
 
     '''
-    import gquant.dataframe_flow as dff
+    from gquant.dataframe_flow import TaskGraph
 
     task_list = run_params_dict['task_list']
     out_list = run_params_dict['out_list']
 
     replace_spec = run_params_dict['replace_spec']
+    task_graph = TaskGraph(task_list)
```
Comment (avolkov1) [inline notebook/mortgage_e2e_gquant/mortgage_gquant_plugins.py]: Rename `task_list` to `task_spec_list`.

File: notebook/mortgage_e2e_gquant/mortgage_run_workflow_daskdistrib.py [hunk 2]
```diff
@@ -152,13 +137,14 @@ def main():
         TaskSpecSchema.inputs: [
             MortgageTaskNames.dask_mortgage_workflow_runner_task_name
         ],
-        TaskSpecSchema.modulepath: mortgage_lib_module
+        TaskSpecSchema.filepath: mortgage_lib_module
     }
 
     task_list = [mortgage_workflow_runner_task, dxgb_trainer_task]
```
Comment (avolkov1) [inline notebook/mortgage_e2e_gquant/mortgage_run_workflow_daskdistrib.py]: Rename `task_list` to `task_spec_list`.

File: notebook/mortgage_e2e_gquant/mortgage_run_workflow_local.py [hunk 3]
```diff
@@ -85,26 +81,27 @@ def main():
...
-        TaskSpecSchema.modulepath: mortgage_lib_module
+        TaskSpecSchema.filepath: mortgage_lib_module
     }
 
-    task_list = [mortgage_workflow_runner_task, xgb_trainer_task]
+    task_spec_list = [mortgage_workflow_runner_task, xgb_trainer_task]
 
     # out_list = [MortgageTaskNames.mortgage_workflow_runner_task_name]
     # ((mortgage_feat_df_pandas, delinq_df_pandas),) = \
     #     dff.run(task_list, out_list)
```
Comment (avolkov1) [inline notebook/mortgage_e2e_gquant/mortgage_run_workflow_local.py]: Delete this commented out snippet.

File: notebook/mortgage_e2e_gquant/mortgage_common.py [hunk 4]
```diff
@@ -46,88 +46,88 @@ def mortgage_etl_workflow_def(
...
         TaskSpecSchema.conf: dict(),
         TaskSpecSchema.inputs: [
             MortgageTaskNames.final_perf_delinq_task_name,
             MortgageTaskNames.load_acqdata_task_name
         ],
-        TaskSpecSchema.modulepath: mortgage_lib_module
+        TaskSpecSchema.filepath: mortgage_lib_module
     }
 
     task_list = [
```
Comment (avolkov1) [inline notebook/mortgage_e2e_gquant/mortgage_common.py]: Rename from `task_list` to `task_spec_list`.

File: notebook/mortgage_e2e_gquant/mortgage_common.py [hunk 5]
```diff
@@ -35,7 +35,7 @@ def mortgage_etl_workflow_def(
         dict specifies a task per TaskSpecSchema.
```
Comment (avolkov1) [inline notebook/mortgage_e2e_gquant/mortgage_common.py]: Update to comment to "Each dict is a task-spec per TaskSpecSchema."

**Comment by yidong72** [comment]
> In the `mortgage_run_workflow_daskdistrib.py` and the notebook, because of deepcopy of conf when I was passing the dask client through the conf:
> 
> ```
>     mortgage_workflow_runner_task = {
>         TaskSpecSchema.task_id:
>             MortgageTaskNames.dask_mortgage_workflow_runner_task_name,
>         TaskSpecSchema.node_type: 'DaskMortgageWorkflowRunner',
>         TaskSpecSchema.conf: {
>             'mortgage_run_params_dict_list': mortgage_run_params_dict_list,
>             'client': client,  # DASK CLIENT CANNOT BE PICKLED I.E. DEEPCOPY BREAKS
>             'use_rmm': use_rmm,
>             'filter_dask_logger': filter_dask_logger,
>         },
>         TaskSpecSchema.inputs: [],
>         TaskSpecSchema.filepath: mortgage_lib_module
>     }
> ```
> 
> I was getting:
> 
> ```
> Traceback (most recent call last):
>   File "/home/avolkov/projects/gitrepos/cuquant/gquant/dataframe_flow/task.py", line 28, in __init__
>     copy.deepcopy(task_spec[TaskSpecSchema.conf])
>   File "/home/avolkov/progs/python_installs/miniconda3/envs/py36-rapids/lib/python3.6/copy.py", line 150, in deepcopy
>     y = copier(x, memo)
>     . . .
> TypeError: can't pickle _asyncio.Future objects
> ```
> 
> I changed the code by passing the dask scheduler address, just a string, and
> creating a new client.
> 
> That caused another error:
> 
> ```
>   File "mortgage_gquant_plugins.py", line 1341, in process
>     dmat_delayed_persist = dmat_delayed.persist()
> ValueError: Inputs contain futures that were created by another client.
> ```
> 
> I could implement this by having the dask client be an output from the first
> task in the task-graph, and passed as an input to the next task. The simplest thing
> to do though is just to not deepcopy the conf fields.

I agree. Let's removed the deep copy lines (L25-28) in the task.py. Since each task graph run needs to build the task graph from scratch, I think we are relatively safe here.
And we should delete "def __setitem__(self, key, value):" method to make it hard to be mutated.

**Pull request merged**